src/sentry/api/endpoints/release_thresholds/utils/get_new_issue_counts.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List
+
+from django.db.models import CharField, Count, Q, Value
+
+from sentry.models.group import Group
+
+if TYPE_CHECKING:
+    from sentry.api.endpoints.release_thresholds.types import EnrichedThreshold
+
+
+def get_new_issue_counts(
+    organization_id: int, thresholds: List[EnrichedThreshold]
+) -> dict[str, int]:
+    """
+    constructs a query for each threshold, filtering on project
+    """
+    queryset = None
+    for t in thresholds:
+        query = Q(
+            project__organization__id=organization_id,
+            project__id=t.project_id,
+        )
+        if t.environment:
+            # If we're filtering on environment, just use the first_release/first_seen from the groupenvironment so we don't need to cross reference tables
+            query &= Q(
+                groupenvironment__first_release__version=t.release,
+                groupenvironment__first_seen__range=(t.start, t.end),
+                groupenvironment__environment__name=t.environment.get("name"),
+            )
+        else:
+            query &= Q(
+                first_release__version=t.release,
+                first_seen__range=(t.start, t.end),
+                groupenvironment__isnull=True,
```

Comment:
Do you need the isnull here? I'd have thought you wanted this just for any environment?

Reply:
I think if there's no environment specified in the release object already, then the user didn't explicitly choose an environment, thus the column remained null, which is why we want to make sure we're still grabbing the release (even though the environment wasn't specified)

Reply:
This is forcing you to only choose groups that explicitly have no environment associated with them though, rather than groups associated with any environment or no environment

Reply:
Also fwiw, I think that every group is associated with at least one env

Reply:
Ahhh true, this is explicit, you're right

Reply:
Yup - was debating this too

This is sort of a feature limitation/issue currently.
Releases currently do not require an environment. So release thresholds should account for releases with no environment set.

there's currently no implementation for _all_ environments either. Just per environment (including None)

So for now - in the case where environment is _not_ provided, i'm explicitly checking for when environment `isnull`

Reply:
https://redash.getsentry.net/queries/5856/source

Every group other than those 1600 has an environment, so you're going to filter to nothing most of the time here.

Reply:
I think that's fine.

I'll need to update the threshold UX so users will expect the `None` environment case - OR update to have either `None` and `All` options 🤔